gulpfile.js

```diff
@@ -5,30 +5,121 @@
    its own file in gulp/tasks. Any files in that directory get
    automatically required below.
    To add a new task, simply add a new task file the gulp/tasks directory.
-   gulp/tasks/default.js specifies the default set of tasks to run
-   when you run `gulp`. */
+   To ignore tasks in production environments, add to the ignoreTasks array. */
 
+const envvars = require( './config/environment' ).envvars;
+const fancyLog = require( 'fancy-log' );
+const fs = require( 'fs' );
+const glob = require( 'glob' );
 const gulp = require( 'gulp' );
-const requireDir = require( 'require-dir' );
+const path = require( 'path' );
 
-// Require all tasks in gulp/tasks, including subfolders.
-requireDir( './gulp/tasks', { recurse: true } );
+// Define tasks that should be ignored in production environment.
+const TASK_PATH = './gulp/tasks/';
+let ignoreTasks = [];
 
-gulp.task( 'build',
-  gulp.series(
+if ( envvars.NODE_ENV === 'production' ) {
+  ignoreTasks = [
+    TASK_PATH + 'audit.js',
+    TASK_PATH + 'docs.js',
+    TASK_PATH + 'lint.js',
+    TASK_PATH + 'test-unit.js'
+  ];
+}
+
+/**
+ * Require a gulp task and log the task to the console.
+ * @param  {string} taskPath - Path to a gulp task.
+ */
+function requireTask( taskPath ) {
+  fancyLog( 'Requiring task', taskPath );
+  // eslint-disable-next-line global-require
+  require( path.resolve( taskPath ) );
+}
+
+/**
+ * Recursively check that a file exists on disk, given a file pattern in the
+ * format file-name.
+ * E.g. Take the file pattern test-unit-scripts.
+ * This function will check (recursively)
+ * whether test-unit-scripts, test-unit, and test exist.
+ * If one of those exist, it will return the first occurence,
+ * otherwise it will return undefined.
+ * @param  {string} filePattern A filename pattern in the format file-name.
+ * @returns {string|undefined} File name that exists, otherwise undefined.
+ */
+function fileExists( filePattern ) {
+  let UNDEFINED;
+  if ( !filePattern ) {
+    return UNDEFINED;
+  }
+  const checkFile = `${ TASK_PATH }${ filePattern }.js`;
+  // eslint-disable-next-line no-sync
+  const doesFileExist = fs.existsSync( checkFile );
+  if ( doesFileExist ) {
+    return checkFile;
+  }
+  const newFile = filePattern.split( '-' );
+  newFile.pop();
+  return fileExists( newFile.join( '-' ) );
+}
+
+/*
+Check for command-line flag option (such as `lint` in `gulp lint`).
+If the option maps to a task in /gulp/tasks/, we can skip loading other tasks.
+*/
+const cliOption = process.argv.slice( 2 );
+let taskName;
+if ( cliOption.length > 0 ) {
+  taskName = cliOption[0].replace( /:/g, '-' );
+}
+const taskFile = fileExists( taskName );
+
+if ( taskFile ) {
+  requireTask( taskFile );
+} else {
```

Comment:
It's difficult to follow where this ends, is it possible either to break out at L:79 or to move the contents here into it's own function and only execute that function within this half of the condition?

Reply:
Fixed

Reply:
Did you push that change?

Reply:
blargh, check now